test/framework/src/test/java/org/elasticsearch/test/VersionUtilsTests.java

```diff
@@ -0,0 +1,167 @@
+        Tuple<List<Version>, List<Version>> t = VersionUtils.resolveReleasedVersions(TestMinorBranch.CURRENT, TestMinorBranch.class);
+        List<Version> released = t.v1();
+        List<Version> unreleased = t.v2();
+        assertEquals(Arrays.asList(TestMinorBranch.V_5_3_0, TestMinorBranch.V_5_3_1, TestMinorBranch.V_5_4_0), released);
+        assertEquals(singletonList(TestMinorBranch.V_5_3_2), unreleased);
+    }
+
+    static class TestNextBranch {
+        public static final Version V_5_3_0 = Version.fromId(Version.V_5_3_0_ID);
+        public static final Version V_5_3_1 = Version.fromId(Version.V_5_3_1_ID);
+        public static final Version V_5_3_2 = Version.fromId(Version.V_5_3_2_ID);
+        public static final Version V_5_4_0 = Version.fromId(Version.V_5_4_0_ID);
+        public static final Version V_5_5_0 = Version.fromId(5050099);
+        public static final Version CURRENT = V_5_5_0;
+    }
+    public void testResolveReleasedVersionsForNextBranch() {
+        Tuple<List<Version>, List<Version>> t = VersionUtils.resolveReleasedVersions(TestNextBranch.CURRENT, TestNextBranch.class);
+        List<Version> released = t.v1();
+        List<Version> unreleased = t.v2();
+        assertEquals(Arrays.asList(TestNextBranch.V_5_3_0, TestNextBranch.V_5_3_1, TestNextBranch.V_5_5_0), released);
+        assertEquals(Arrays.asList(TestNextBranch.V_5_3_2, Version.V_5_4_0), unreleased);
+    }
+
+    static class TestAlphaBranch {
+        public static final Version V_5_3_0 = Version.fromId(Version.V_5_3_0_ID);
+        public static final Version V_5_3_1 = Version.fromId(Version.V_5_3_1_ID);
+        public static final Version V_5_3_2 = Version.fromId(Version.V_5_3_2_ID);
+        public static final Version V_5_4_0 = Version.fromId(Version.V_5_4_0_ID);
+        public static final Version V_6_0_0_alpha1 = Version.fromId(Version.V_6_0_0_alpha1_ID);
+        public static final Version V_6_0_0_alpha2 = Version.fromId(Version.V_6_0_0_alpha2_ID);
+        public static final Version CURRENT = V_6_0_0_alpha2;
+    }
+    public void testResolveReleasedVersionsForAlphaBranch() {
+        Tuple<List<Version>, List<Version>> t = VersionUtils.resolveReleasedVersions(TestAlphaBranch.CURRENT, TestAlphaBranch.class);
+        List<Version> released = t.v1();
+        List<Version> unreleased = t.v2();
+        assertEquals(Arrays.asList(TestAlphaBranch.V_5_3_0, TestAlphaBranch.V_5_3_1, TestAlphaBranch.V_6_0_0_alpha1,
+                TestAlphaBranch.V_6_0_0_alpha2), released);
+        assertEquals(Arrays.asList(TestAlphaBranch.V_5_3_2, TestAlphaBranch.V_5_4_0), unreleased);
+    }
+
+    /**
+     * Test that the released inference logic produces expected "unreleased"
+     * versions. This test will fail every time we create a new version
+     * constant and it should be upgraded by hand. Take this opportunity to
+     * make sure that it lines up with reality.
+     */
+    public void testUnreleaseCurrent() {
```

Comment:
I'll drop it. I had thought it'd be a nice reminder for the next few releases *until* we automate the version bumping.